File: model/payment.go
package model

type PaymentOrderStatus string

const (
	PaymentOrderStatusPending PaymentOrderStatus = "pending"
	PaymentOrderStatusSuccess PaymentOrderStatus = "success"
	PaymentOrderStatusClosed  PaymentOrderStatus = "closed"
)

// PaymentOrder 用户充值订单。
type PaymentOrder struct {
	ID              string             `json:"id" gorm:"primaryKey"`
	UserID          string             `json:"userId" gorm:"index"`
	Credits         int                `json:"credits"`
	AmountCents     int                `json:"amountCents"`
	PaymentProvider string             `json:"paymentProvider"`
	PaymentMethod   string             `json:"paymentMethod"`
	ProviderTradeNo string             `json:"providerTradeNo"`
	Status          PaymentOrderStatus `json:"status" gorm:"index"`
	NotifyPayload   string             `json:"notifyPayload" gorm:"type:text"`
	CreatedAt       string             `json:"createdAt"`
	UpdatedAt       string             `json:"updatedAt"`
	PaidAt          string             `json:"paidAt"`
}

type PaymentOrderList struct {
	Items []PaymentOrder `json:"items"`
	Total int            `json:"total"`
}

File: web/src/services/api/payment.ts
import { apiGet, apiPost, compactApiParams } from "@/services/api/request";

export type PaymentMethod = {
    type: string;
    name: string;
    enabled: boolean;
};

export type PaymentOrderStatus = "pending" | "success" | "closed";

export type PaymentOrder = {
    id: string;
    userId: string;
    credits: number;
    amountCents: number;
    paymentProvider: string;
    paymentMethod: string;
    providerTradeNo: string;
    status: PaymentOrderStatus;
    createdAt: string;
    updatedAt: string;
    paidAt: string;
};

export type PaymentOrderListResponse = {
    items: PaymentOrder[];
    total: number;
};

export type CreateEpayOrderResponse = {
    order: PaymentOrder;
    paymentUrl: string;
};

export async function createEpayOrder(token: string, payload: { credits: number; method: string }) {
    return apiPost<CreateEpayOrderResponse>("/api/payments/epay/orders", payload, token);
}

export async function fetchPaymentOrders(token: string, query: { page?: number; pageSize?: number } = {}) {
    return apiGet<PaymentOrderListResponse>("/api/payments/orders", compactApiParams(query), token);
}

export async function fetchPaymentOrder(token: string, id: string) {
    return apiGet<PaymentOrder>(`/api/payments/orders/${encodeURIComponent(id)}`, undefined, token);
}
